### PR TITLE
Switch to cakephp/twig-view 1.0

### PR DESCRIPTION
**Pull request by othercorey**

### CODE REVIEW

**Comment by ADmad** [comment]
After switch to the new twigview plugin do we release a new minor for bake or new major?

**Comment by dereuromark** [comment]
A minor should be fine, as it will auto pull the needed dependency.

**Comment by othercorey** [comment]
The only problem with a minor now is we won't be 100% backwards compatible with cakephp/twig-view. The namespaces will change. I don't think there's anything else we're going to change that will affect end users.

**Comment by dereuromark** [comment]
For a require dev dependency thats probably okish.

**Comment by othercorey** [comment]
Can we marked this as replacing wyrihaximus/twig-view in packagist? Will that throw an error for projects that don't update their own require dependency if they use TwigView outside of bake?

**Comment by dereuromark** [comment]
If you use `conflict` you can do that. It can then only install once you remove the other twig dependency.

**Comment by ADmad** [comment]
I think we should use `conflict`, otherwise there will be problems for those who use twigview directly.

**Comment by dereuromark** [comment]
Now that we have a beta, we should switch to the cake one
It otherwise blocks other plugins from switching

    - cakephp/bake dev-master requires wyrihaximus/twig-view ^5.0.1 -> satisfiable by wyrihaximus/twig-view[5.0.1].
    - cakephp/bake dev-master requires wyrihaximus/twig-view ^5.0.1 -> satisfiable by wyrihaximus/twig-view[5.0.1].
    - cakephp/twig-view 1.0.0-beta1 conflicts with wyrihaximus/twig-view[5.0.1].
    - cakephp/twig-view 1.0.0-beta1 conflicts with wyrihaximus/twig-view[5.0.1].
    - dereuromark/cakephp-dto dev-cake-twig requires cakephp/twig-view ^1.0.0 -> satisfiable by cakephp/twig-view[1.0.0-beta1].
    - Installation request for dereuromark/cakephp-dto dev-cake-twig as 1.0.0 -> satisfiable by dereuromark/cakephp-dto[dev-cake-twig].
    - Installation request for cakephp/bake dev-master as 2.0.5 -> satisfiable by cakephp/bake[dev-master].

File: composer.json [hunk 1]
```diff
@@ -41,6 +41,8 @@
             "Cake\\Test\\": "./vendor/cakephp/cakephp/tests/"
         }
     },
+    "prefer-stable": true,
+    "minimum-stability": "dev",
```
Comment (ADmad) [inline composer.json]:
```suggestion
    "minimum-stability": "beta",
```

File: composer.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
         "php": ">=7.2",
         "cakephp/cakephp": "^4.0",
         "cakephp/plugin-installer": "^1.0",
-        "wyrihaximus/twig-view": "^5.0.1"
+        "cakephp/twig-view": "^1.0.0"
```
Comment (ADmad) [inline composer.json]:
```suggestion
        "cakephp/twig-view": "^1.0"
```

Comment (dereuromark) [reply composer.json]: Why? 3 digit is the default, and same meaning :)

**Comment by dereuromark** [comment]
The conflict should not go here, though. Right?
It should go to twig-view repo as there we need to make composer aware it cannot be used together with the legacy one.

**Comment by ADmad** [comment]
> The conflict should not go here, though. Right?

Yes, it has been added to `cakephp/twig-view`.

**Comment by othercorey** [comment]
If we switch to the beta, it requires everyone to move their minimum stability again. Not sure we want to do that.

**Comment by dereuromark** [comment]
Then release a stable plugin version now.
At this point it is as stable as the other one before - and we can add patches for things that are broken.

**Comment by othercorey** [comment]
Can no one try it in a project somewhere once?

**Comment by dereuromark** [comment]
I already did as part of Dto plugin (see link), and the sandbox using it from project level.
But I didn't use it as project only templating. Maybe someone else can
Or you can add a demo of it to the sandbox.

**Comment by ADmad** [comment]
I reverted back to use `dev-master` of TwigView. Given the flurry of changes we are making to TwigView it's better to keep testing against bake regularly, instead of waiting till next beta to discover bugs/BC breaks.

**Comment by ADmad** [comment]
As I had suspected there were issues which I have fixed.

**Comment by othercorey** [comment]
There are missing variable exceptions in the unit tests with `strict_variables => true`.

**Comment by dereuromark** [comment]
prefix + traitName
not too bad

**Comment by othercorey** [comment]
I could not fix this error:

> 1) Bake\Test\TestCase\Command\TemplateCommandTest::testGetContentWithRoutingPrefix
Twig\Error\RuntimeError: Impossible to access an attribute ("type") on a null variable.
>
> /cake/templates/bake/element/form.twig:52

The test is faking data that always loads the $modelObject. However, setting `modelObject` to the model instance causes another failure. I'm not sure if is valid.

File: templates/bake/Template/add.twig [hunk 3]
```diff
@@ -19,4 +19,4 @@
  * @var \{{ entityClass }} ${{ singularVar }}
  */
 ?>
-{% element 'form' %}
+{{ element('form') -}}
```
Comment (ADmad) [inline templates/bake/Template/add.twig:22]: Whitespacing consistency should be maintained between the add and edit templates.

Here trailing whitespace is removed using `-}}` while it's not in edit.twig.

Comment (othercorey) [reply templates/bake/Template/add.twig:22]: It shouldn't be needed. I accidentally added a newline and was trying to fix it. Thought I updated both.

**Comment by ewbarnard** [comment]
File `docs/en/development.rst`, section "Bake Template Syntax" links to twig 2.x, saying that Bake uses the Twig 2.x template syntax. `cakephp/twig-view` uses current Twig (3.x), correct? Given that we link to current Twig at the top of `development.rst` we could probably revert the link to plain text "Bake template files use the Twig template syntax."

**Comment by othercorey** [comment]
Updated to official 1.0 release.

File: templates/bake/Template/index.twig [hunk 4]
```diff
@@ -39,7 +39,7 @@
                 <tr>
 {% for field in fields %}
 {% set isKey = false %}
-{% if associations.BelongsTo %}
+{% if associations.BelongsTo is defined %}
```
Comment (ADmad) [inline templates/bake/Template/index.twig:42]: Turning on `strict_variables` by default for Twig will potentially cause errors in user templates. In order to keep upgrade experience smooth for users we should consider turning is off for bake.

Comment (othercorey) [reply templates/bake/Template/index.twig:42]: They are only enabled in debug which is turned on for unit tests. They shouldn't experience it locally ... right?

Comment (ADmad) [reply templates/bake/Template/index.twig:42]: bake is always used locally with debug on.

**Comment by othercorey** [comment]
- Added `Bake.twigStrictVariables` config that's set by unit tests
- Updated controller.twig to generates @property/@method tags that pass phpcs spacing.

**Pull request merged**